kafka/channel/pkg/apis/messaging/v1alpha1/kafka_channel_conversion.go

```diff
@@ -0,0 +1,178 @@
+/*
+Copyright 2020 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"context"
+	"fmt"
+	"knative.dev/eventing-contrib/kafka/channel/pkg/apis/messaging/v1beta1"
+	eventingduckv1 "knative.dev/eventing/pkg/apis/duck/v1"
+	eventingduckv1alpha1 "knative.dev/eventing/pkg/apis/duck/v1alpha1"
+	eventingduckv1beta1 "knative.dev/eventing/pkg/apis/duck/v1beta1"
+	"knative.dev/pkg/apis"
+	duckv1 "knative.dev/pkg/apis/duck/v1"
+	duckv1alpha1 "knative.dev/pkg/apis/duck/v1alpha1"
+)
+
+// ConvertTo implements apis.Convertible.
+// Converts channel (from v1alpha1.KafkaChannel) into v1beta1.KafkaChannel
+func (source *KafkaChannel) ConvertTo(ctx context.Context, obj apis.Convertible) error {
+	switch sink := obj.(type) {
+	case *v1beta1.KafkaChannel:
+		statusAddressable := &duckv1.Addressable{
+			URL: nil,
+		}
+		if source.Status.AddressStatus.Address != nil {
+			if err := source.Status.AddressStatus.Address.ConvertTo(ctx, statusAddressable); err != nil {
+				return err
+			}
+		}
+
+		subscribableStatus := eventingduckv1.SubscribableStatus{
+			Subscribers: nil,
+		}
+		if source.Status.SubscribableStatus != nil && len(source.Status.SubscribableStatus.Subscribers) > 0 {
+			subscribableStatus.Subscribers = make([]eventingduckv1.SubscriberStatus, len(source.Status.SubscribableStatus.Subscribers))
+			for i, ss := range source.Status.SubscribableStatus.Subscribers {
+				subscribableStatus.Subscribers[i] = eventingduckv1.SubscriberStatus{
+					UID:                ss.UID,
+					ObservedGeneration: ss.ObservedGeneration,
+					Ready:              ss.Ready,
+					Message:            ss.Message,
+				}
+			}
```

Comment:
There's some code in this PR that can be moved to `ConvertTo/From` methods of e.g. `SubscribableStatus`.
But we jump from `v1alpha1` to `v1` for these and no conversion between those versions is available yet: https://github.com/knative/eventing-contrib/blob/c826903acee0c1649d3646c50f83fb15d85363df/vendor/knative.dev/eventing/pkg/apis/duck/v1alpha1/subscribable_types_conversion.go#L115

I might address that later.

Reply:
Might be worth having the conversion for SubscribableStatus in a separate func here so that its easier to move / refactor later?